basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/AddElementToListNodeType.java

```diff
@@ -0,0 +1,58 @@
+package club.mondaylunch.gatos.basicnodes;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import club.mondaylunch.gatos.core.data.DataBox;
+import club.mondaylunch.gatos.core.data.DataType;
+import club.mondaylunch.gatos.core.data.ListDataType;
+import club.mondaylunch.gatos.core.graph.connector.NodeConnector;
+import club.mondaylunch.gatos.core.graph.type.NodeType;
+
+public class AddElementToListNodeType extends NodeType.Process {
+
+    @Override
+    public Map<String, DataBox<?>> settings() {
+        return Map.of();
+    }
+
+    @Override
+    public Set<NodeConnector.Input<?>> inputs(UUID nodeId, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        return Set.of(
+            new NodeConnector.Input<>(nodeId, "list", ListDataType.GENERIC_LIST),
+            new NodeConnector.Input<>(nodeId, "element", DataType.ANY)
+        );
+    }
+
+    @Override
+    public Set<NodeConnector.Output<?>> outputs(UUID nodeId, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        return Set.of(
+            new NodeConnector.Output<>(nodeId, "output", inputTypes.get("list"))
+        );
+    }
+
+    @Override
+    public Map<String, CompletableFuture<DataBox<?>>> compute(Map<String, DataBox<?>> inputs, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        if (inputTypes.get("element").listOf() != inputTypes.get("list")) {
+            throw new IllegalArgumentException("The Element's Type does not match that of the List");
+        }
+        return Map.of(
+            "output", CompletableFuture.completedFuture(this.getGenericListBox(
+                Stream.concat(
+                    DataBox.get(inputs, "list", ListDataType.GENERIC_LIST).orElse(List.of()).stream(),
+                    Stream.of(DataBox.get(inputs, "element", DataType.ANY).orElseThrow()))
+                .collect(Collectors.toList()),
```

Comment:
Please just copy the list and add the element this is mildly confusing to read and is probably many times slower.